Add handler for ICNS files (wxBITMAP_TYPE_ICON)

TcT2k opened the pull request:
After some recent changes bitmaps and icons with the type
`wxBITMAP_TYPE_ICON` could no longer be loaded. This implements a handler
for this (macOS only) bitmap type like the handler for
`wxBITMAP_TYPE_ICON_RESOURCE`.

I'm not sure which changes in `src/osx/core/bitmap.cpp` caused this error, but I think in 3.1.1 I had no problem loading files with this type. I've also come accross `src/osx/carbon/icon.cpp` which seems unused? It should probably be removed if I'm not mistaken.

TcT2k committed:
Add handler for ICNS files (wxBITMAP_TYPE_ICON)

After some recent changes bitmaps and icons with the type
wxBITMAP_TYPE_ICON could no longer be loaded. This implements a handler
for this (macOS only) bitmap type like the handler for
wxBITMAP_TYPE_ICON_RESOURCE.

vadz commented:
I have no idea about how did it work before, but the new code looks mostly fine to me (being perfectionist, I'd refactor `wxICNSResourceHandler::LoadFile()` to reuse the common part and add a simple test checking we can load e.g. `src/osx/carbon/wxmac.icns`, but well...), thanks!

I think this also addresses [this bug](http://trac.wxwidgets.org/ticket/18647) as `wxIcon` is the same as `wxBitmap` under Mac, right?

csomor reviewed (approved):
Thanks Tobias, this looks fine to me

TcT2k commented:
> I have no idea about how did it work before

I also couldn't quite figure out how it worked before.

> 
> I think this also addresses [this bug](http://trac.wxwidgets.org/ticket/18647) as `wxIcon` is the same as `wxBitmap` under Mac, right?

Yes `wxIcon` just wraps `wxBitmap` in the code I saw. The issue described in the SO thread sounds exactly like the issue I had, so it should also be fixed.

TcT2k committed:
Remove unused carbon files

These files where no longer referenced by any build system
and where probably left overs from before carbon was removed.
